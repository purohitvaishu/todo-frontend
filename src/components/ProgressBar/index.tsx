/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable no-unused-vars */
/* eslint-disable @typescript-eslint/no-unused-vars */
import React from "react";
import CircularProgress, {
  CircularProgressProps
} from "@material-ui/core/CircularProgress";
import Box from "@material-ui/core/Box";

type Progress = {
  props?: CircularProgressProps;
  value: number;
};

const CircularProgressWithLabel: React.FC<Progress> = (
  props: CircularProgressProps & { value: number }
) => {
  const { value } = props;

  return (
    <Box
      position="relative"
      display="inline-flex"
      style={{ margin: "auto 0", marginRight: 12 }}
    >
      <CircularProgress
        variant="determinate"
        style={{ color: "lightgrey" }}
        size={30}
        thickness={4}
        {...props}
        value={100}
      />
      <CircularProgress
        style={{ position: "absolute", color: "#F76363" }}
        size={30}
        variant="determinate"
        {...props}
      />
    </Box>
  );
};

export default CircularProgressWithLabel;
