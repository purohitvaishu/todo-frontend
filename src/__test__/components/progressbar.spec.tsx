/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { shallow } from "enzyme";
import ProgressBar from "../../components/ProgressBar";

describe("Test: Bucket", () => {
  test("container", () => {
    const props = {
      value: 20
    };

    const result = shallow(<ProgressBar {...props} />);
    expect(result.debug()).toMatchSnapshot();
  });
});
