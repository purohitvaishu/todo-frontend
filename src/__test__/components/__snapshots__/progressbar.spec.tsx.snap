// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: Bucket container 1`] = `
"<Styled(MuiBox) position=\\"relative\\" display=\\"inline-flex\\" style={{...}}>
  <WithStyles(ForwardRef(CircularProgress)) variant=\\"determinate\\" style={{...}} size={30} thickness={4} value={100} />
  <WithStyles(ForwardRef(CircularProgress)) style={{...}} size={30} variant=\\"determinate\\" value={20} />
</Styled(MuiBox)>"
`;
